src/lib.rs

```diff
@@ -0,0 +1,1159 @@
+						Ok(())
+					}
+					Err(e) => {
+						log_error!(sync_logger, "Sync of Lightning wallet failed: {}", e);
+						Err(e)
+					}
+				}
+			})
+		})?;
+
+		Ok(())
+	}
+
+	/// Close a previously opened channel.
+	pub fn close_channel(
+		&self, channel_id: &[u8; 32], counterparty_node_id: &PublicKey,
+	) -> Result<(), Error> {
+		self.peer_store.remove_peer(counterparty_node_id)?;
+		match self.channel_manager.close_channel(channel_id, counterparty_node_id) {
+			Ok(_) => Ok(()),
+			Err(_) => Err(Error::ChannelClosingFailed),
+		}
+	}
+
+	/// Send a payement given an invoice.
+	pub fn send_payment(&self, invoice: Invoice) -> Result<PaymentHash, Error> {
+		if self.running.read().unwrap().is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let mut outbound_payments_lock = self.outbound_payments.lock().unwrap();
+
+		let payment_hash = PaymentHash((*invoice.payment_hash()).into_inner());
+		let payment_secret = Some(*invoice.payment_secret());
+
+		match lightning_invoice::payment::pay_invoice(
+			&invoice,
+			Retry::Timeout(LDK_PAYMENT_RETRY_TIMEOUT),
+			self.channel_manager.as_ref(),
+		) {
+			Ok(_payment_id) => {
+				let payee_pubkey = invoice.recover_payee_pub_key();
+				let amt_msat = invoice.amount_milli_satoshis().unwrap();
+				log_info!(self.logger, "Initiated sending {} msats to {}", amt_msat, payee_pubkey);
+
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: payment_secret,
+						status: PaymentStatus::Pending,
+						amount_msat: invoice.amount_milli_satoshis(),
+					},
+				);
+
+				Ok(payment_hash)
+			}
+			Err(payment::PaymentError::Invoice(e)) => {
+				log_error!(self.logger, "Failed to send payment due to invalid invoice: {}", e);
+				Err(Error::InvalidInvoice)
+			}
+			Err(payment::PaymentError::Sending(e)) => {
+				log_error!(self.logger, "Failed to send payment: {:?}", e);
+
+				outbound_payments_lock.insert(
+					payment_hash,
+					PaymentInfo {
+						preimage: None,
+						secret: payment_secret,
+						status: PaymentStatus::Failed,
+						amount_msat: invoice.amount_milli_satoshis(),
+					},
+				);
+				Err(Error::PaymentFailed)
+			}
+		}
+	}
+
+	/// Send a payement given a so-called "zero amount" invoice, i.e., an invoice that leaves the
+	/// amount paid to be determined by the user.
+	pub fn send_adjustable_value_payment(
+		&self, invoice: Invoice, amount_msat: u64,
+	) -> Result<PaymentHash, Error> {
+		if self.running.read().unwrap().is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let mut outbound_payments_lock = self.outbound_payments.lock().unwrap();
+
+		let payment_hash = PaymentHash((*invoice.payment_hash()).into_inner());
+		let payment_secret = Some(*invoice.payment_secret());
+
+		if invoice.amount_milli_satoshis().is_some() {
+			log_error!(
+				self.logger,
+				"Failed to pay the given invoice: expected \"zero-amount\" invoice, please use send_payment."
+			);
+			return Err(Error::InvalidInvoice);
+		}
```

Comment:
Would we want to allow paying an invoice with amount over the stated amount? (e.g. tip) Or maybe we don't support that in LDK?